jumpscale/sals/nginx/nginx.py

```diff
@@ -318,16 +326,45 @@ def generate_certificates(self, retries=6):
         if self.domain:
             if self.key_path and self.cert_path and self.fullchain_path:
                 # only use install command if an existing key and certificate were set
-                cmd = self.certbot.install_cmd
+                self.install_certifcate(retries=retries)
             else:
-                cmd = self.certbot.run_cmd
-
-            for _ in range(retries):
-                rc, out, err = j.sals.process.execute(cmd)
-                if rc > 0:
-                    j.logger.error(f"Generating certificate failed {out}\n{err}")
-                else:
-                    break
+                self.obtain_and_install_certifcate(retries=retries)
+
+    def install_certifcate(self, retries=6):
+        """Construct and Execute install certificate command
+        Alternative to certbot install
+
+        Args:
+            retries (int, optional): Number of retries Certbot will try to install the certificate if failed. Defaults to 6.
+        """
+        cmd = self.certbot.install_cmd
+        j.logger.debug(f"Execute: {' '.join(cmd)}")
+        for _ in range(retries):
```

Comment:
do we have a generic way to do retries? if not we can add a function or a decorator that can do this and reuse it in both `install_certificate` and `obtain_and_install_certificate`.

Reply:
also, I think we can remove this retries here, we did it for run command because the CA server might timeout...etc, in case of installation, this shouldn't happen.

Reply:
Do we still need the function or decorator after using it in `obtain_and_install_certificate` only?

Reply:
I think no, we don't need it.